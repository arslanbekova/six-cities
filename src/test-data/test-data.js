export const AuthorizationStatus = {
  AUTH: 'AUTH',
  NO_AUTH: 'NO_AUTH',
};

export const AuthInfo = {
  AUTHORIZED_USER: {
    email: 'mail@domain.com',
    avatarUrl: './img/avatar.png',
    id: 1,
    isPro: true,
    name: 'Mary'
  },
  UNAUTHORIZED_USER: {
    email: null
  }
};
