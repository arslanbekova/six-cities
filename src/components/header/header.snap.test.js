import React from  'react';
import { Router } from 'react-router-dom';
import { createMemoryHistory } from 'history';
import { render } from '@testing-library/react';
import * as reactRedux from 'react-redux'
import Header from './header';
import { AuthorizationStatus, AuthInfo } from '../../test-data/test-data';

describe(`Should Header render correctly`, () => {
  const useSelectorMock = jest.spyOn(reactRedux, 'useSelector');
  const useDispatchMock = jest.spyOn(reactRedux, 'useDispatch');
  const history = createMemoryHistory();

  beforeEach(() => {
    useSelectorMock.mockClear()
    useDispatchMock.mockClear()
  });

  test(`if user is authorized`, () => {
    useSelectorMock.mockReturnValue({
      authorizationStatus: AuthorizationStatus.AUTH,
      authInfo: AuthInfo.AUTHORIZED_USER
    });

    const dummyDispatch = jest.fn();
    useDispatchMock.mockReturnValue(dummyDispatch);
    expect(dummyDispatch).not.toHaveBeenCalled();

    const { container } = render(
      <Router history={history}>
        <Header/>
      </Router>
    );
    expect(container).toMatchSnapshot();
  })
  test(`if user is unauthorized`, () => {
    useSelectorMock.mockReturnValue({
      authorizationStatus: AuthorizationStatus.NO_AUTH,
      authInfo: AuthInfo.UNAUTHORIZED_USER
    });

    const dummyDispatch = jest.fn();
    useDispatchMock.mockReturnValue(dummyDispatch);
    expect(dummyDispatch).not.toHaveBeenCalled();

    const { container } = render(
      <Router history={history}>
        <Header/>
      </Router>
    );
    expect(container).toMatchSnapshot();
  })
})
