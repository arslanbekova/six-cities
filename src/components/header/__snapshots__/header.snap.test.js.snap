// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should Header render correctly if user is authorized 1`] = `
<div>
  <header
    class="header"
  >
    <div
      class="container"
    >
      <div
        class="header__wrapper"
      >
        <div
          class="header__left"
        >
          <a
            class="header__logo-link header__logo-link--active"
            href="/"
          >
            <img
              alt="6 cities logo"
              class="header__logo"
              height="41"
              src="img/logo.svg"
              width="81"
            />
          </a>
        </div>
        <nav
          class="header__nav"
        >
          <ul
            class="header__nav-list"
          >
            <li
              class="header__nav-item user"
              style="display: flex; align-items: baseline;"
            >
              <a
                class="header__nav-link header__nav-link--profile"
                href="/favorites"
              >
                <div
                  class="header__avatar-wrapper user__avatar-wrapper"
                  style="background-image: url(./img/avatar.png);"
                />
                <span
                  class="header__user-name user__name"
                >
                  mail@domain.com
                </span>
              </a>
              <button
                class="header__logout-button button"
                style="margin-left: 10px; width: 18px; height: 18px; background-image: url(../img/logout.svg);"
                type="button"
              />
            </li>
          </ul>
        </nav>
      </div>
    </div>
  </header>
</div>
`;

exports[`Should Header render correctly if user is unauthorized 1`] = `
<div>
  <header
    class="header"
  >
    <div
      class="container"
    >
      <div
        class="header__wrapper"
      >
        <div
          class="header__left"
        >
          <a
            class="header__logo-link header__logo-link--active"
            href="/"
          >
            <img
              alt="6 cities logo"
              class="header__logo"
              height="41"
              src="img/logo.svg"
              width="81"
            />
          </a>
        </div>
        <nav
          class="header__nav"
        >
          <ul
            class="header__nav-list"
          >
            <li
              class="header__nav-item user"
              style="display: flex; align-items: baseline;"
            >
              <a
                class="header__nav-link header__nav-link--profile"
                href="/favorites"
              >
                <div
                  class="header__avatar-wrapper user__avatar-wrapper"
                  style="background-image: url(undefined);"
                />
                <span
                  class="header__user-name user__name"
                >
                  Sign in
                </span>
              </a>
              <button
                class="header__logout-button button"
                style="margin-left: 10px; width: 18px; height: 18px; background-image: url(../img/logout.svg);"
                type="button"
              />
            </li>
          </ul>
        </nav>
      </div>
    </div>
  </header>
</div>
`;
